.github/workflows/build.yaml

```diff
@@ -251,9 +251,54 @@ jobs:
           path: |
             image/rhel/docs
 
+  check-collector-images-exist:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        tag_suffix: ["", "-fast", "-fast-slim"]
+    steps:
+    - name: Checkout repo
+      uses: actions/checkout@v4
+
+    - name: Get COLLECTOR_VERSION
+      id: collector-version
+      run: |
+        echo "collector-version=$(make --quiet --no-print-directory collector-tag)" >> "${GITHUB_OUTPUT}"
+
+    - name: Check image exists
```

Comment:
It would be possible to add steps that use [tj-actions/changed-files](https://github.com/tj-actions/changed-files), BUT it would require more work to handle both PRs and branch push differently.
Unlikely, though not impossible: the scanner or collector images could be cleaned up by Quay's image retention, which would fail this job and require the engineer to bump the versions.
The job is a prerequisite for image builds and runs in parallel to longer-running compilation jobs, so doesn't add to the overall build time.
There is no monetary cost accrued on GHA for ACS to run this job. 

TLDR: there is no harm to running the check on each build, so why overengineer it.